debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -529,7 +529,7 @@ public Long currentTransactionId() throws SQLException {
     public long currentXLogLocation() throws SQLException {
         AtomicLong result = new AtomicLong(0);
         int majorVersion = connection().getMetaData().getDatabaseMajorVersion();
-        query(majorVersion >= 10 ? "select * from pg_current_wal_lsn()" : "select * from pg_current_xlog_location()", rs -> {
+        query(majorVersion >= 10 ? "select (case pg_is_in_recovery() when 't' then pg_last_wal_receive_lsn() else pg_current_wal_lsn() end) AS pg_current_wal_lsn" : "select * from pg_current_xlog_location()", rs -> {
```

Comment:
This mimics similar implementations elsewhere, e.g. [here](https://github.com/mblp/postgres_exporter/commit/11cfc43fe9326e74ce46b9b9358a5b4ed87be513).